File: yunj/level1/5_an_unfinished_athlete.java
import java.util.HashMap;
import java.util.Set;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";

        HashMap<String, Integer> part = new HashMap<>();

        for (String p : participant) {
            int value = 1;
            if (part.containsKey(p)) {
                value = part.get(p) + 1;
            }
            part.put(p, value);
        }

        for (String c : completion) {
            int value = part.get(c);
            part.put(c, value - 1);
        }

        Set<String> keys = part.keySet();

        for (String key : keys) {
            if (part.get(key) == 1)
                return key;
        }

        return answer;
    }
}

File: sangjin/level1/5_an_unfinished_athlete.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <iostream>

using namespace std;

long long hash(string s){
    const int m = 1e9 + 9; // 적당히 큰 소수...
    vector<long long> p = {1};
    long long hash = 0;

    for(int i = 0; i < s.size(); i++){
        hash = (hash + (s[i] - 'a' + 1) * p[i]) % m;
    }
    return hash;
}

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";

    return answer;
}